### DISCUSSION TITLE
Signifincace of including filename when sending search criteria

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#635] [Q&A] [answered]
What is the significance of including the filename when sending search criteria - see [here](https://github.com/opendcs/opendcs/blob/28673064894655cb319105321cad3f0d6a61bdc8/src/main/java/lrgs/ldds/LddsClient.java#L780). Looks like it runs fine even if you don't include the filename at all.

Also, is there any reference for why the first 40 bytes are the filename and why message data begins after the first 50 bytes?

**Comment by MikeNeilson** [comment]
If the upstream LRGS has a matching crit file (the name) it would used. I don't think it's used much anymore, nor would i recommend upstream LRGS systems allowing users to create files.

**Comment by ghost** [comment]
Thank you!

For the second question - Am I correct in assuming that the first 50 bytes should not be used for actual search criteria data?

**Reply by MikeNeilson** [reply, marked as answer]
https://github.com/opendcs/dcs_standards/blob/main/source/dds-14.rst#transferring-search-criteria-to-the-server

Sorry, forgot that specification is actually defined there for a moment. It's 50 blank characters because it used to be used for a filename, but is not longer used, so all actual search criteria start at byte 51.
